### ISSUE TITLE
Always show personal note upload button

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#15482]
Additionally a button, which is just disabled/greyed out (instead of being completely invisible) because of internal problems can still provide a tool tip or even a popup with the underlying reason for its deactivation. Writing this, the sometimes suddenly vanishing ``Upload personal note`` button comes to my mind.  Would be way better to tell the user that its currently not available because of e. g. connection or authentication problems or because he's no PM than to just hide it, which is often seen as a SW bug.

Mnemonic:  A hidden button can't tell no story.  ;)

_Originally posted by @MagpieFourtyTwo in https://github.com/cgeo/cgeo/issues/15312#issuecomment-2002618929_

**Comment by Lineflyer** [comment]
Same BTW for the personal note upload icon which is hidden for BMs:
It would be better to always show it and provide a feedback like " Uploading your note is only available for PM. Check your membership state or re-login".
Cause quite a few users contact support about this in cases where they are not logged in for some reason....but thats another issue.

From https://github.com/cgeo/cgeo/issues/15312#issuecomment-2002515007

**Comment by ztNFny** [comment]
> Uploading your note is only available for PM. Check your membership state or re-login".

For those situations where the login failed this would be useful - but for all those basic users it'll add several buttons which never can do anything. Kinda feels like the official app which shows you some caches only to then tell you that you can't access them because you're not paying.

Maybe we could remember whether the user was signed in as Premium and thus distuingish failed logins from "real" basic users?

**Comment by MagpieFourtyTwo** [comment]
> Kinda feels like the official app which shows you some caches only to then tell you that you can't access them because you're not paying.

Yep, that's right - nevertheless it can be good to know what c:geo can do for me, if I switch to PM. May be interesting for those who are undecided. BTW: Is there a manual page which shows differences between being PM vs. BM, seen from c:geo's perspective?

However, then let's concentrate on "internal stuff" - from my personal experience, the questions that users usually ask me suggest that it seems to be rather difficult for a normal user to find the more advanced features of c:geo. Most of them don't even think about that there could be "more" than just clicking caches on a map (which usually even is online only).  But what surprised me the most was that most of them did not even know about c:geo's capability to navigate. Until I saw a few days ago that - although offline routing is mentioned in the "Basic" view of the settings - "Basic" gives no possibility to activate it. So how should they get an idea?

Hence I would suggest to let c:geo advertise its functions a bit more. Which may not really fit into this issue, but was *actually* the idea behind the "greyed out upload button". Greyed out options or a toast here or there with a hint, how things may go even better, e. g. when routing to a target, but navigation is not configured ... TBC

**Comment by Lineflyer** [comment]
> BTW: Is there a manual page which shows differences between being PM vs. BM, seen from c:geo's perspective?

There is a page, not sure if it might be outdated:
https://manual.cgeo.org/de/basicmembers

**Comment by Lineflyer** [comment]
> Yep, that's right - nevertheless it can be good to know what c:geo can do for me, if I switch to PM. 

The main reason for raising this issue was not to do promotion for premium memberships but to make users aware of not being logged in. Alternatively we would need to find a good solution to present offline/not logged in status in general or buffer the membership status to avoid such temporary flaws, which users don't understand ("I am premium but cannot upload my notes").

**Comment by RJCheuk** [comment]
I want to work this issue :)

**Comment by MagpieFourtyTwo** [comment]
AFAIK you can only see the ``PN Upload`` button (which would then indeed show up left of the ``Edit`` button) if you're Premium.
Cause AFAIK base members are not allowed to have PNs at all, anyway.

BTW: If shown, the button's icon will disappear while c:geo is uploading the content (frame lines remain visible) - really good solution, IMO.  Would like to have this for the ``Update`` button on the ``Details`` page, too ... but will create an own FR for this.  ;)

**Comment by RJCheuk** [comment]
Hi there, just to update my progross and ask questions abou the test.

**Always show personal note upload button has been accomplished.**

*I am basic user and before that I could not see the button, and now it's always on
Before:

<img width="414" alt="Image" src="https://github.com/user-attachments/assets/cf82aa70-63c1-45dd-9b32-25c139a91ca1" />

After:
<img width="414" alt="Image" src="https://github.com/user-attachments/assets/8dba414e-88fd-4f5a-a204-d0fd1597ca15" />

**Additionally, I have added features to remind users of the button's availability（grayed out the button）with toasts, reminding them why they couldn't use it.**

1.  For basic user with log-in normally, it shows "Only Premium Members can upload personal notes."

<img width="414" alt="Image" src="https://github.com/user-attachments/assets/b644d73b-e471-4584-b2a0-cbd6394b5ed2" />

2. For connectorPN == null, it shows "Personal note upload not supported on this site."(string changed before this screenshot)

<img width="414" alt="Image" src="https://github.com/user-attachments/assets/a315778e-dfb2-41c0-a191-cf365b6f7fd6" /> 

3.For offline users, it shows " Currently offline – cannot upload personal notes."

<img width="469" alt="Image" src="https://github.com/user-attachments/assets/20936fc2-b448-4e1f-ab0a-7bed2a6fef7a" />


Continue work: 

1. Do I need to cover this scenario? 
_Not logged in → “Please log in to upload personal notes.”_
I don't think it's useful, users might know they are offline before entering this page. Please tell me what you think to help wity my work.

2. How can I test this scenario?
_Online and with permission → Button is highlighted, click to directly trigger the upload logic_
Becasue I am basic user and I cannot take this test to cover this scenario. But I don't think it could be problem for premuim users.

Thank you for you help and guidance, look forward to your reply!

**Comment by MagpieFourtyTwo** [comment]
Looks good to me.

WRT ``Not logged in → “Please log in to upload personal notes.”``: I think it's worth considering, depends on how effortful the implementation is.  Cause although login is performed automatically by c:geo, this may not work if you're in a region with weak GSM signal (or none at all).  c:geo then takes a while, just displaying "Logging in ..." on the home screen - and most of the users I know immediately skip this screen without even looking at the text, thus do not even notice that they're not logged in.  Moreover c:geo does not give a message if login fails resp. if it has given up on retries - if it gives up at all ... maybe someone may answer this who knows more about c:geo's retry strategy if first login tries fail ...

BSo, if it gives up some time, the user may have no (obvious) possibility to login other than restarting c:geo, thus the message may then be a bit more informative on what to do ... perhaps like ``Not logged in, check your login state`` or so ...?

**Comment by ztNFny** [comment]
I'd simplify all those failing cases to a single one with "Personal note upload unavailable. Premium members only. Are you logged in?"
It's not as helpful but having tons of special cases in the code for such a simple thing feels wrong to me.

**Comment by MagpieFourtyTwo** [comment]
> How can I test ``Online and with permission → Button is highlighted, click to directly trigger the upload logic``

I would suggest to provide a build to a developer who is Premium, so he can test it in his simulator.
